suricata/update/main.py

```diff
@@ -1018,6 +1018,26 @@ def check_output_directory(output_dir):
                 "Failed to create directory %s: %s" % (
                     output_dir, err))
 
+def check_version(suricata_version):
+    index_filename = sources.get_index_filename()
+    if not os.path.exists(index_filename):
+        logger.warning("No index exists, will use bundled index.")
+        logger.warning("Please run suricata-update update-sources.")
+    index = sources.Index(index_filename)
+    version = index.get_versions()
+    if suricata_version.full in version['suricata']['recommended'] or \
+            "dev" in str(suricata_version.full):
+        logger.info("Suricata version %s is up to date", suricata_version.full)
+    elif suricata_version.short in version['suricata'] and \
+            suricata_version.full not in version['suricata']['4.1']:
```

Comment:
I'm surprised to see a hardcoded version here. Shouldn't this be fully handled by the index file?

We know the running Suricata version, so we can take '4.1.4', reduce it to '4.1', look up the '4.1' entry in the index and check if our 4.1.4 matches the version we have for '4.1'.

Reply:
Thanks for catching this. Going to fix this issue.

Reply:
Could you please suggest what should be done in case if the Suricata version is greater than the recommended version. Like if someone uses the following command:
`sudo ./bin/suricata-update --check-versions --suricata-version 5.2.3`

Reply:
@vagishagupta23 I think the recommended version will always be the latest stable (in our intel index and that is kind of our responsibility to manage). So unless its dev or pre-release, this situation should not arise.
For example, currently, the stable release and the recommended version is `4.1.4` (check here: https://github.com/OISF/suricata/releases/tag/suricata-4.1.4) however there can be some pre-releases like there is `5.0.0-beta1` now but this is not the recommended version because its not stable.

I think it would be helpful for you to have a look at the previous releases and understand the difference and naming schemes: https://github.com/OISF/suricata/releases

If required, we can always change our index to accomodate some more information. ;)

Reply:
> This makes me realize that `suricata-version` functionality is probably not working fine? No matter what version I provide, it says in the logs "Forcing Suricata version to 5.6". 5.6 does not even exist. Seems like it just gives the output without checking if that Suricata version exists on the system. @jasonish is this the expected behavior?

More or less.  This is to force a specific version for the case where you are using Suricata-Update without having Suricata installed, or building a ruleset for a different version of Suricata that you have installed.

One use case is running Suricata-Update on one system, then using some deployment/provisioning system like Ansible to push the rules out to your Suricata installations.

Reply:
OK. Thanks! :)

Reply:
> @vagishagupta23 I think the recommended version will always be the latest stable (in our intel index and that is kind of our responsibility to manage). So unless its dev or pre-release, this situation should not arise.
> For example, currently, the stable release and the recommended version is `4.1.4` (check here: https://github.com/OISF/suricata/releases/tag/suricata-4.1.4) however there can be some pre-releases like there is `5.0.0-beta1` now but this is not the recommended version because its not stable.
> 
> I think it would be helpful for you to have a look at the previous releases and understand the difference and naming schemes: https://github.com/OISF/suricata/releases
> 
> If required, we can always change our index to accomodate some more information. ;)

Thanks. These links would be helpful to understand the naming schemes. :)